src/ape/contracts/base.py

```diff
@@ -677,6 +677,74 @@ def _mutable_methods_(self) -> Dict[str, ContractTransactionHandler]:
             # NOTE: Must raise AttributeError for __attr__ method or will seg fault
             raise AttributeError(str(err)) from err
 
+    def call_view_method(self, method_name: str, *args, **kwargs) -> Any:
+        """
+        Call a contract's view function directly using the method_name.
+        This is helpful in the scenario where the contract has a
+        method name matching an attribute of the BaseAddress class,
+        such as 'nonce' or 'balance'
+
+        Args:
+            method_name (str): The contract method name to be called
+            *args: Contract method arguments.
+            **kwargs: Transaction values, such as ``value`` or ``sender``
+
+        Returns:
+            Any: Output of smart contract view call.
+
+        """
+
+        if method_name in self._view_methods_:
```

Comment:
We could create a helper method to share some of this code between the two new methods, but see some of the other feedback before doing that!

It may look like this:

```python
def _call_contract(method_name: str, *args, **kwargs):`
```

Reply:
We could do this, it comes down to how we want the `call_view_methods()` function to behave. Right now I have it that the `_view_methods_` handler in `call_view_methods()` function does NOT use `.transact`. This is because `.transact` requires the user to define the `sender` `**kwarg`. View functions imo shouldn't require defining the `sender` because state isn't being changed. 

If we did add the `.transact` to the handler in `call_view_methods()` then we could share the code between the two functions

Reply:
Okay great, thats how I implemented it in the latest changes :D Given the difference between the functions, I don't think it makes sense to use helper code here, unless i'm missing something.